packages/terra-overlay/src/Overlay.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import Portal from './Portal';
+import OverlayManager from './OverlayManager';
+
+import './Overlay.scss';
+
+const ESCAPE = 27;
+const overlayManager = new OverlayManager();
+
+const propTypes = {
+  /**
+   * Whether or not <Overlay/> should be rendered
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Set full screen overlay
+   */
+  isFullScreen: PropTypes.bool,
+  /**
+   * DOM element that will be overlaid
+   */
+  container: PropTypes.object,
+  /**
+   * Callback function for closing the overlay
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Allow closing overlay by clicking backdrop
+   */
+  canClickOutsideClose: PropTypes.bool,
+  /**
+   * Center the content horizontally and vertically
+   */
+  centerContent: PropTypes.bool,
+  /**
+   * Visible overlay content when overlay is open
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  isOpen: false,
+  isFullScreen: true,
+  container: document.body,
+  centerContent: true,
+  canClickOutsideClose: true,
+};
+
+class Overlay extends React.Component {
+  constructor() {
+    super();
+    this.state = {};
```

Comment:
I don't see state referenced anywhere in this file, this can probably be removed